### PR TITLE
[DOCS] uni-directional CCR disaster recovery

### LINKED ISSUE #84854: [Docs] Step-by-step tutorial for uni-directional CCR failover
### Description

As of writing, ccr does not offer automatic failover.  Can we please add the following tutorial for the failover scenario?

The initial setup can be skipped as it's similar to [Tutorial: Set up cross-cluster replication](https://github.com/elastic/elasticsearch/edit/8.1/docs/reference/ccr/getting-started.asciidoc). Adding it here for completeness.

# Initial setup (uni-directional CCR with `DR` cluster following `Production` cluster) 
### Step1: Create remote clusters on `DR` and point to `production`
```
### On DR cluster ###
PUT /_cluster/settings
{
  "persistent" : {
    "cluster" : {
      "remote" : {
        "production" : {
          "seeds" : [
            "127.0.0.1:9300" 
          ]
        }
      }
    }
  }
}
```

### Step2: Create an index on `Production`
```
### On Production cluster ###
PUT /my_index
POST /my_index/_doc/1
{
  "foo":"bar"
}
```
### Step3: Create follower index on DR
```
### On DR cluster ###
PUT /my_index/_ccr/follow 
{ 
  "remote_cluster" : "production", 
  "leader_index" : "my_index" 
}
```

### Step4: Test follower index on DR
```
### On DR cluster ###
GET /my_index/search

### This should show up the content created on production (foo/bar)
```

⚠️ Ingestion should only be written to the `Production` cluster, all search queries can be directed to either `Production` or `DR` clusters.

# When Production down:
### Step2: On the Elasticsearch side, turn the follower indices in the `DR` into regular indices:
Ensure no writes are occurring on the leader index (if the data centre is down, or cluster is unavailable, no action needed)
On DR: Convert the follower index to a normal index in Elasticsearch (capable of accepting writes)

```
### On DR cluster ###
POST /my_index/_ccr/pause_follow
POST /my_index/_close           
POST /my_index/_ccr/unfollow    
POST /my_index/_open
```
### Step3: On the Client side, manually re-enable ingestion of my_index to the `DR` cluster. (You can test that the index should be writable:
```
### On DR cluster ###
POST my_index/_doc/2
{
  "foo": "new"
}  
```
⚠️ Make sure all traffic is redirected to the `DR` cluster during this time.

# Once the `Production` comes back:
### Step2: Create remote clusters on `Production` and points to `DR`
```
### On Production cluster ###
PUT _cluster/settings
{
  "persistent" : {
    "cluster" : {
      "remote" : {
        "dr" : {
          "seeds" : [
            "127.0.0.2:9300" 
          ]
        }
      }
    }
  }
}
```
### Step3: Create follower indices in `Production`, connecting them to the leader in `DR`. The former leader indices in `Production` have outdated data and will need to be discarded/deleted.  Wait for Production follower indices to catch up.  Once it is caught up, you can turn the follower indices in `Production` to regular index again.
```
### On Production cluster ###
DELETE my_index

### Create follower index on Production to follow from DR cluster
PUT /my_index/_ccr/follow 
{ 
  "remote_cluster" : "dr", 
  "leader_index" : "my_index" 
}

### Wait for my_index to catch up with DR and contain all the documents.
GET my_index/_search

### Stop following from DR to turn my_index into a regular index.
POST /my_index/_ccr/pause_follow
POST /my_index/_close
POST /my_index/_ccr/unfollow
POST /my_index/_open 
```
### Step4: Delete the former `DR` writeable indices that contain outdated data now. Create follower indices in the `DR`  again to ensure that all changes from `Production` are streamed to `DR`. (This is the same as the initial setup)
```
### On DR cluster ###
DELETE my_index

### Create follower index on `DR` to follow from the `Production` cluster
PUT /my_index/_ccr/follow 
{ 
  "remote_cluster" : "production", 
  "leader_index" : "my_index" 
}
```
### Step5: On the Client side, manually re-enable ingestion to the `Production` cluster.

⚠️ Ingestion should only be written onto `Production`, all search queries can be directed to either `Production` or `DR` clusters.

### PR DESCRIPTION
**Pull request by Leaf-Lin**
Detailed step-by-step tutorial for uni-directional CCR disaster recovery

closes: https://github.com/elastic/elasticsearch/issues/84854

### CODE REVIEW

**Comment by henningandersen** [comment]
@elasticmachine run elasticsearch-ci/docs
@elasticmachine run elasticsearch-ci/docs-check

**Review by henningandersen — COMMENTED** [review]
You will need to include this new page in order to make it visible (and to be able to preview it).

**Comment by Leaf-Lin** [comment]
@elasticmachine run elasticsearch-ci/docs
@elasticmachine run elasticsearch-ci/docs-check

**Comment by lockewritesdocs** [comment]
I added this file to the table of contents so that it displays in the navigation. I also made changes to the introductory section, but I have questions about how this tutorial works. We make a lot of assumptions in this document. For example, the first step says:

> >1. On the Clients side, pause ingestion of `my_index` into `prod1`.
> 
> TIP: Ensure no writes are occurring on the leader index (if the data centre is
> down, or cluster is unavailable, no action needed).

How do you pause ingestion? Through Kibana? If so, where? What is "the Clients side"? How can I tell if writes are occurring on the leader index or check the cluster's availability?

I don't think that we can assume users have all of this knowledge and instinctually know where to complete these tasks. We should be much more prescriptive and provide both direction (such as where to go in the Cloud console) and examples.

File: docs/reference/ccr/disaster-recovery.asciidoc [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+[[ccr-disaster-recovery-tutorial]]
+=== Tutorial: Configure disaster recovery with {ccr}
+++++
+<titleabbrev>Configure disaster recovery</titleabbrev>
+++++
+
+////
+[source,console]
+----
+PUT /server-metrics
```
Comment (lockewritesdocs) [inline docs/reference/ccr/disaster-recovery.asciidoc:11]: I don't understand why we're setting up this index if it's not referenced at all in the document. Should `my_index` in the later examples instead use `server-metrics`?

**Comment by Leaf-Lin** [comment]
I've made some changes to this doc including clarifying how to pause ingest on the client side and adding an image to clarify the 3 scenarios.

File: docs/reference/ccr/disaster-recovery.asciidoc [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+
+Learn how to failover and failback between two clusters based on
+uni-directional {ccr}. The following tutorial is designed for mutable indices
+(data need to be updated after write). If you have append-only data (logs,
+metrics, observability, and security events), you should follow
+https://www.elastic.co/blog/bi-directional-replication-with-elasticsearch-cross-cluster-replication-ccr[Bi-directional
+replication with Elasticsearch cross-cluster replication (CCR) blog] to set up
+bi-directional {ccr} which automatically failover and failback without human
+intervention.
```
Comment (Leaf-Lin) [inline docs/reference/ccr/disaster-recovery.asciidoc:16]:
```suggestion
uni-directional {ccr}. You can also visit <<ccr-disaster-recovery-bi-directional-tutorial,ccr bi-directional tutorial>> to set up replicating data streams that automatically failover and failback without human
intervention.
```

File: docs/reference/ccr/index.asciidoc [hunk 3]
```diff
@@ -308,6 +308,7 @@ and use {ccr}:
 * <<ccr-upgrading>>
 
 include::getting-started.asciidoc[]
+include::disaster-recovery.asciidoc[]
```
Comment (Leaf-Lin) [inline docs/reference/ccr/index.asciidoc:311]:
```suggestion
include::uni-directional-disaster-recovery.asciidoc[]
```

**Comment by Leaf-Lin** [comment]
Closing this in favour of https://github.com/elastic/elasticsearch/pull/91491

**Pull request closed without merging**